### ISSUE TITLE
Biome cannot provide downstream a11y linting for Pragma components

### ISSUE THREAD

**Issue by Trivaxy** [canonical/pragma#1096]
We're setting up our CI/CD pipeline in Debcrafters Workflows, and we're currently evaluating our options regarding formatting/linting.

At the moment, the plan we had is to use Biome for formatting and *general* linting. The issue is that Biome cannot do full a11y linting for downstream consumers of Pragma, as it does not support component mapping.

Essentially, within the Pragma repo itself, Biome will do a11y linting because e.g. for the `<Button>` component it has access to the raw DOM that includes the underlying `<button>`, which it understands and can lint.

For a downstream consumer (like us), Biome only sees `<Button>` without knowing anything about its internals and thus skips linting it entirely. Our original plan was to bring in ESLint with the `eslint-plugin-jsx-a11y` plugin, and set up a component mapping, so that ESLint understands that, for example, `<Button>` is semantically a `<button>` and should be linted as if it were one. It essentially fills the hole Biome has due to Biome not letting you map Pragma components this way.

We also investigated different tooling, mainly Oxlint. Like ESLint, it supports component mapping, and it also has the same functionality as `eslint-plugin-jsx-a11y`, so it can be set up to handle all of the linting on its own. The only thing we'd need to do is pair it with a formatter such as Biome (or Oxfmt, but that one's in beta).

The question here is what Pragma can do to help standardize how downstream consumers handle linting Pragma components in general. It ships Biome by default, which won't do full a11y linting for Pragma components when used downstream, meaning essentially every person using Pragma would need to accept either having Pragma components unlinted on their end or needing a workaround (like the one I described with ESLint).
